pyhf/constraints.py

```diff
@@ -0,0 +1,131 @@
+from . import get_backend, default_backend
+
+class gaussian_constraint_combined(object):
+    def __init__(self,pdfconfig):
+        tensorlib, _ = get_backend()
+        self.tensorlib_name = tensorlib.name
+        self.prepped = False
+        # iterate over all constraints order doesn't matter....
+        
+        self.par_indices = list(range(len(pdfconfig.suggested_init())))
+        self.data_indices = list(range(len(pdfconfig.auxdata)))
+        self.mod_and_slice = [
+            (pdfconfig.modifier(cname),pdfconfig.par_slice(cname))
+            for cname in pdfconfig.auxdata_order
+        ]
+
+    def _precompute(self):
+        tensorlib, _ = get_backend()
+        # did things change that we need to recompute?
+        tensor_type_change = tensorlib.name != self.tensorlib_name
+        if (not tensor_type_change) and self.prepped:
```

Comment:
This logic is kinda odd to me. You're precomputing if the tensor type changes or if not prepped. You can get rid of prepped if you initialize the tensor type to "None" and call `_precompute()` in the `__init__()` which catches a tensor type change. Technically, it doesn't matter if you're prepped or not if the tensor type changes.